rosbag2_cpp/src/rosbag2_cpp/typesupport_helpers.cpp

```diff
@@ -116,8 +116,11 @@ get_typesupport(
   auto library_path = get_typesupport_library_path(package_name, typesupport_identifier);
 
   try {
-    library = std::make_shared<rcpputils::SharedLibrary>(library_path);
-
+    // TODO(karsten1987) This logic is pretty fragile.
+    // The library should not be passed in via a reference to a shared pointer.
```

Comment:
So should we just fix the interface now?

Reply:
I was thinking about splitting the function into two parts. One which initializes the library and a second which extracts the type support handle from it. 

What do you think?

Reply:
That sounds good to me - I feel like it makes sense to handle it immediately rather than put a todo - unless it's significantly more work

Reply:
I'll try to get to this by the end of today.